ruby/spec/helpers/headers.rb

```diff
@@ -0,0 +1,17 @@
+def prepare_auth_token
+  creds = []
+  
+  File.foreach('config/.access') do |line|
+    creds.append(line.split('=').last.strip)
+  end
+
+   Base64.encode64(creds.join(':'))
+end
+
+def prepare_auth_header
+  {'HTTP_AUTHORIZATION' => "Basic #{prepare_auth_token}" }
+end
+
+def prepare_content_type_header
+  {'CONTENT_TYPE' => 'application/json'}
+end
```

Comment:
Okay so this is organizes the code even better, but I noticed one thing right away: Branching

There is an increased branching behavior due to the switch and if we were writing low-level, high-performance code we would have to avoid it. 

But I will try it out :)